internal/nightfallconfig/nightfall_config.go

```diff
@@ -9,72 +9,66 @@ import (
+	MaxConcurrentRoutinesCap = 50
+	DefaultMaxNumberRoutines = 20
 
-var (
-	one                           int32 = 1
-	confidencePossible                  = nightfallAPI.CONFIDENCE_POSSIBLE
-	nightfallDetectorType               = nightfallAPI.DETECTORTYPE_NIGHTFALL_DETECTOR
-	nightfallAPIKey                     = nightfallAPI.NIGHTFALLDETECTORTYPE_API_KEY
-	nightfallAPIKeyName                 = string(nightfallAPI.NIGHTFALLDETECTORTYPE_API_KEY)
-	nightfallCryptographicKey           = nightfallAPI.NIGHTFALLDETECTORTYPE_CRYPTOGRAPHIC_KEY
-	nightfallCryptographicKeyName       = string(nightfallAPI.NIGHTFALLDETECTORTYPE_CRYPTOGRAPHIC_KEY)
-	defaultNightfallConfig              = &NightfallConfigFileStructure{
-		Conditions: []*nightfallAPI.Condition{
-			{
-				Detector: &nightfallAPI.Detector{
-					DetectorType:      &nightfallDetectorType,
-					NightfallDetector: &nightfallAPIKey,
-					DisplayName:       &nightfallAPIKeyName,
+	nightfallConfigFilename      = ".nightfalldlp/config.json"
+	defaultConditionsInfoMessage = "Using default Detection Rule with detectors API_KEY and CRYPTOGRAPHIC_KEY"
+)
+
+var defaultNightfallConfig = &ConfigFile{
+	DetectionRules: []nf.DetectionRule{
+		{
+			Detectors: []nf.Detector{
+				{
+					MinNumFindings:    1,
+					MinConfidence:     nf.ConfidencePossible,
+					DetectorType:      nf.DetectorTypeNightfallDetector,
+					NightfallDetector: "API_KEY",
+					DisplayName:       "API_KEY",
 				},
-				MinNumFindings: &one,
-				MinConfidence:  &confidencePossible,
-			},
-			{
-				Detector: &nightfallAPI.Detector{
-					DetectorType:      &nightfallDetectorType,
-					NightfallDetector: &nightfallCryptographicKey,
-					DisplayName:       &nightfallCryptographicKeyName,
+				{
+					MinNumFindings:    1,
+					MinConfidence:     nf.ConfidencePossible,
+					DetectorType:      nf.DetectorTypeNightfallDetector,
+					NightfallDetector: "CRYPTOGRAPHIC_KEY",
+					DisplayName:       "CRYPTOGRAPHIC_KEY",
 				},
-				MinNumFindings: &one,
-				MinConfidence:  &confidencePossible,
 			},
 		},
-		MaxNumberRoutines: DefaultMaxNumberRoutines,
-	}
-)
+	},
+	MaxNumberRoutines: DefaultMaxNumberRoutines,
+}
 
-// NightfallConfigFileStructure struct representation of nightfall config file
-type NightfallConfigFileStructure struct {
-	ConditionSetUUID   string                    `json:"conditionSetUUID"`
-	Conditions         []*nightfallAPI.Condition `json:"conditions"`
-	MaxNumberRoutines  int                       `json:"maxNumberConcurrentRoutines"`
-	TokenExclusionList []string                  `json:"tokenExclusionList"`
-	FileInclusionList  []string                  `json:"fileInclusionList"`
-	FileExclusionList  []string                  `json:"fileExclusionList"`
+// ConfigFile is the struct of the JSON nightfall config file
+type ConfigFile struct {
+	DetectionRuleUUIDs []uuid.UUID        `json:"detectionRuleUUIDs"`
+	DetectionRules     []nf.DetectionRule `json:"detectionRules"`
```

Comment:
how are you going to manage this breaking change?

Reply:
i have written a script to convert v2 config structs to v3, used them to convert the files in this repo.